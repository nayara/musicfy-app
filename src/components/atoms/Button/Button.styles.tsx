import styled from "styled-components";

const Button = styled.button`
  background: none;
  color: inherit;
  border: none;
  padding: 0.5rem;
  font: inherit;
  cursor: pointer;
  outline: inherit;
`;

export { Button };
