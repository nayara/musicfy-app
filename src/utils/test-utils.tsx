import { RenderOptions, render } from "@testing-library/react";
import { FC, ReactElement, ReactNode } from "react";
import { Providers } from "src/providers";

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <Providers>{children}</Providers>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
