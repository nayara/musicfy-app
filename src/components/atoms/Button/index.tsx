import { FC, ReactNode } from "react";
import * as Style from "./Button.styles";

type TButtonProps = {
  children: ReactNode;
  onClick: () => void;
};

const Button: FC<TButtonProps> = ({ children, onClick }) => {
  return <Style.Button onClick={onClick}>{children}</Style.Button>;
};

export default Button;
