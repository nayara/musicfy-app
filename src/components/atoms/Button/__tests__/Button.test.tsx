import { userEvent } from "@testing-library/user-event";
import { render, screen } from "../../../../utils/test-utils";
import Button from "..";

describe("<Button />", () => {
  it("should render button correctly", async () => {
    const mockFunction = jest.fn();

    const { container } = render(
      <Button onClick={mockFunction}>Click me!</Button>
    );

    expect(container).toMatchSnapshot();

    const buttonElement = screen.getByRole("button", { name: "Click me!" });

    await userEvent.click(buttonElement);

    expect(mockFunction).toHaveBeenCalled();
  });
});
