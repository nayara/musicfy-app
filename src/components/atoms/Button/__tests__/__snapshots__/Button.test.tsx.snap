// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render button correctly 1`] = `
.c0 {
  background: none;
  color: inherit;
  border: none;
  padding: 0.5rem;
  font: inherit;
  cursor: pointer;
  outline: inherit;
}

<div>
  <button
    class="c0"
  >
    Click me!
  </button>
</div>
`;
